### ISSUE TITLE
`yarn opensearch` arg to setup dependencies 

### ISSUE THREAD

**Issue by kavilla** [opensearch-project/OpenSearch-Dashboards#2544]
**Is your feature request related to a problem? Please describe.**

As a developer working on an external plugin (for example the AD dashboards plugin) that requires a corresponding OpenSearch plugin, but I can not use the `yarn opensearch` command to start up the cluster because it will install a `min` of OpenSearch.

Therefore, I must either download a docker image, download a distribution, or pull down OpenSearch and clone my OpenSearch repo and build start. Even if I only want to do OpenSearch Dashboards development.

**Describe the solution you'd like**

I would like to run the `yarn opensearch` command to setup up a data source with an OpenSearch plugin that my OpenSearch Dashboards plugin requires. 

For example, if I would like to start a snapshot of OpenSearch from OpenSearch Dashboards. I run `yarn opensearch snapshot` to get a `min` snapshot.

But I would like to do something like 

`yarn opensearch snapshot -P anomaly-detection`
or
`yarn opensearch snapshot -P file:{path}/anomaly-detection.zip`
or 
`yarn opensearch snapshot -P {URL}/anomaly-detection.zip`

Which will download the snapshot and install security plugin into the snapshot of OpenSearch. So that I can work completely in OpenSearch Dashboards.

We can already configure the cluster config by passing `-E` [[REF](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/packages/osd-opensearch/src/cli_commands/snapshot.js#L49)]. So adding more to this file that then executes [opensearch-plugin](https://github.com/opensearch-project/OpenSearch/blob/main/distribution/src/bin/opensearch-plugin) with the value passed in.

**Describe alternatives you've considered**

Add a new config to the [plugin manifest](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/opensearch_dashboards.json#L6). That tells what data source requirement it needs and as a developer if I add this configuration then the start up OpenSearch installs those plugins if available.

**Additional context**

n/a

**Comment by joshuarrrr** [comment]
@kavilla Am I right in thinking this is mostly helpful in the case where a developer is just working on an OpenSearch Dashboard plugin? If they're working on both the OpenSearch Dashboard and OpenSearch plugin at the same time, they wouldn't be able to take advantage of the snapshot, right?

**Comment by manasvinibs** [comment]
@joshuarrrr I think you're right! As part of the Extension installation process, we want to solve some of the problems for OpenSearch Dashboard developers and this will be one of them.

**Comment by kavilla** [comment]
@joshuarrr yes but using just using the CLI . Developers can download with knowing the snapshot link. The goal here though should be focusing on the OpenSearch Dashboards developer to only pull down OpenSearch Dashboards requirements.
